resources/codegen_inputs/os2.rs

```diff
@@ -0,0 +1,59 @@
+#![parse_module(read_fonts::tables::os2)]
+
+/// [`os2`](https://docs.microsoft.com/en-us/typography/opentype/spec/os2)
+table Os2 {
+    /// The version: 0x00005000 for version 0.5, 0x00010000 for version 1.0.
+    #[version]
+    #[compile(self.compute_version())]
+    version: BigEndian<Version16Dot16>,
+    x_avg_char_width: BigEndian<i16>,
+    us_weight_class: BigEndian<u16>,
+    us_width_class: BigEndian<u16>,
+    fs_type: BigEndian<u16>,
+    y_subscript_x_size: BigEndian<i16>,
+    y_subscript_y_size: BigEndian<i16>,
+    y_subscript_x_offset: BigEndian<i16>,
+    y_subscript_y_offset: BigEndian<i16>,
+    y_superscript_x_size: BigEndian<i16>,
+    y_superscript_y_size: BigEndian<i16>,
+    y_superscript_x_offset: BigEndian<i16>,
+    y_superscript_y_offset: BigEndian<i16>,
+    y_strikeout_size: BigEndian<i16>,
+    y_strikeout_position: BigEndian<i16>,
+    s_family_class: BigEndian<i16>,
+    panose_10: BigEndian<u8>,
+    ul_unicode_range_1: BigEndian<u32>,
+    ul_unicode_range_2: BigEndian<u32>,
+    ul_unicode_range_3: BigEndian<u32>,
+    ul_unicode_range_4: BigEndian<u32>,
+    ach_vend_id: BigEndian<Tag>,
+    fs_selection: BigEndian<u16>,
+    us_first_char_index: BigEndian<u16>,
+    us_last_char_index: BigEndian<u16>,
+    s_typo_ascender: BigEndian<i16>,
+    s_typo_decender: BigEndian<i16>,
+    s_typo_line_gap: BigEndian<i16>,
+    us_win_ascent: BigEndian<u16>,
+    us_win_descent: BigEndian<u16>,
+
+    #[available(Version16Dot16::VERSION_1_0)]
```

Comment:
Looking at the spec, it seems the OS/2 version field is actually just a `uint16`. We should be able to replace all these versions with raw integers (and the version field with `u16` as in `COLR`)

Reply:
oh yea, good catch. That means that this code was also getting incorrect values everywhere, since all fields after the version would've been shifted by 2 bytes.

In any case this obviates the need for any custom logic, since our implementation of `Comparable` for `u16` does what we will need.

Reply:
It sounds like you need either a range or set of versions instead of just one?

Reply:
I made it a u16.

Reply:
I think this is resolved.